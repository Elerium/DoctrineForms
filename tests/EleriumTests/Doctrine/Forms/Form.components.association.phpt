<?php

/**
 * Part of Elerium Framework
 * Copyright (c) 2013
 */
 
namespace EleriumTests\Doctrine\Forms;

use Elerium;
use EleriumTests\Mocks;
use Tester\Assert;
use Elerium\Doctrine\Forms;

require_once __DIR__ . '/../../../bootstrap.php';
require_once __DIR__ . '/../../../Mocks/EntityManager.php';
require_once __DIR__ . '/entities.php';

$entityManager = new Mocks\EntityManager;

$form = new Forms\Form($entityManager, new Entity\Person);

Assert::throws(function() use ($form) {
	$form->addAssociation('unknown');
}, 'Elerium\InvalidArgumentException', "Field 'unknown' in entity 'EleriumTests\\Doctrine\\Forms\\Entity\\Person' does not exist.");

Assert::throws(function() use ($form) {
	$form->addAssociation('name');
}, 'Elerium\InvalidArgumentException', "Field 'name' in entity 'EleriumTests\\Doctrine\\Forms\\Entity\\Person' is not association field.");

Assert::throws(function() use ($form) {
	$form->addAssociation('childrens');
}, 'Elerium\InvalidArgumentException', "Field 'childrens' in entity 'EleriumTests\\Doctrine\\Forms\\Entity\\Person' is not single valued association field.");

$form->addText('id');
$form->addText('name');
$parent = $form->addAssociation('parent');
$parent->addText('id');
$parent->addText('name');
$parent2 = $parent->addAssociation('parent');
$parent2->addText('id');
$parent2->addText('name');

Assert::true($parent instanceof Forms\Container);
Assert::true($parent2 instanceof Forms\Container);
Assert::same(file_get_contents(__DIR__ . '/Form.components.association.expected'), $form->__toString());