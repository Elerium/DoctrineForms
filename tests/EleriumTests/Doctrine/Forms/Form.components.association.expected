<form action="" method="post">

<table>
<tr class="required">
	<th><label class="required" for="frm-id"></label></th>

	<td><input type="text" class="text" name="id" id="frm-id" required="required" data-nette-rules="{op:':integer',msg:'Please enter a numeric value.'},{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>

<tr class="required">
	<th><label class="required" for="frm-name"></label></th>

	<td><input type="text" class="text" name="name" id="frm-name" required="required" data-nette-rules="{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>

<tr class="required">
	<th><label class="required" for="frm-parent-id"></label></th>

	<td><input type="text" class="text" name="parent[id]" id="frm-parent-id" required="required" data-nette-rules="{op:':integer',msg:'Please enter a numeric value.'},{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>

<tr class="required">
	<th><label class="required" for="frm-parent-name"></label></th>

	<td><input type="text" class="text" name="parent[name]" id="frm-parent-name" required="required" data-nette-rules="{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>

<tr class="required">
	<th><label class="required" for="frm-parent-parent-id"></label></th>

	<td><input type="text" class="text" name="parent[parent][id]" id="frm-parent-parent-id" required="required" data-nette-rules="{op:':integer',msg:'Please enter a numeric value.'},{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>

<tr class="required">
	<th><label class="required" for="frm-parent-parent-name"></label></th>

	<td><input type="text" class="text" name="parent[parent][name]" id="frm-parent-parent-name" required="required" data-nette-rules="{op:':filled',msg:'Please complete mandatory field.'}" value="" /></td>
</tr>
</table>

</form>
